11-Backbonejs/06-Adding-Custom-Events.md

```diff
@@ -0,0 +1,255 @@
+    "click": "onClick"
+  },
+  onClick: function() {
+    console.log("Clicked!  Model:  " + this.model.get("title"));
+  },
+  render: function() {
+    this.$el.html(this.model.get("title"));
+    return this;
+  }
+});
+``` 
+
+If you wanted to listen for a specific item within the view you could, with jQuery, specify the item using a CSS Selector.  For example if there was a block within the view with a class `title` and you wanted to listen for clicks on that block, you could put this line into the events hash `"click .title": "onTitleClick"`.  This would cause the View to listen for clicks on any element within the view and with the `title` class.  When any sub-element with that class is clicked on the `onTitleClick` event handler will run.
+
+
+Backbone Events API allows you to add event handlers with the `.on` method as you saw earlier.  This lets you create custom events that can be triggered in your Models or Views to update other objects when something occurs, a change in status, user interaction etc.  
+
+Looking at the TaskViews in the Codepen, we can see when a task is clicked on using a DOM event.  However it would be REALLY useful if we could update another view when a Todo item is clicked on, for example to show more details on the selected Todo item.  Thus we will return to the Backbone Events API and the idea of an Event Bus later.
+
+### listenTo 
+
+Similar to the `.on` function, the `.listenTo()` method lets a Backbone object listen to an event that occurs on another object.  Look below and see how `listenTo()` differs from the `on()` function.  The main difference between the two is the context in which they add callbacks.  `on()` lets you directly add a callback function to another object.  `listenTo()` tells an object to listen for events on another object.  This allows the 'listener' to keep track of the objects for which it is listening.  
+
+Here's another example:
+
+```javascript
+// A view listening to a Task Model's event.
+this.listenTo(this.task, 'change:completed', this.filterOne);
+this.task.on('change:completed', this.filterOne);
+```
+
+Notice that the `listenTo()` method has the listening object subscribe to another object's event.  The on method has the model, which generates the event remember to notify the callback function.  There's another difference between `listenTo()` and `on()`.  If you call another function `stopListening` any listeners added by `listenTo()` will be removed.  Similarly if a view is removed with the `remove` method all listeners using `listenTo()` will be removed.  Models behave the same way, models deleted with the `destroy` method will call `stopListening` on any subscribers using `listenTo()`  In the **old** days, the only way to subscribe to an event was with the `on()` function.  That caused problems because when a view or model was removed the callback functions were still listening for events, which wasted memory and lead to unexpected behaviors.    
```

Comment:
* There's a whole lot going on in this paragraph, it'd probably be best to move the information about how/when to stop listening into its own paragraph.
* I'm not sure that the final couple of sentences about the old days is meaningful information for the students. If there's a technique to avoid those problems we should probably inform them of that technique explicitly.
* The second sentence "The on method has the model..." is confusing to me, I'm not sure what it's saying.